internal/services/cdn/cdn_frontdoor_rule_sets_association_resource.go

```diff
@@ -0,0 +1,223 @@
+package cdn
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/cdn/mgmt/2021-06-01/cdn" // nolint: staticcheck
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+const cdnFrontDoorRuleSetAssociationResourceType string = "azurerm_cdn_frontdoor_rule_sets_association"
+
+func resourceCdnFrontDoorRuleSetAssociation() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceCdnFrontDoorRuleSetAssociationCreate,
+		Read:   resourceCdnFrontDoorRuleSetAssociationRead,
+		Delete: resourceCdnFrontDoorRuleSetAssociationDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceIdThen(func(id string) error {
+			_, err := parse.FrontDoorRuleSetAssociationID(id)
+			return err
+		}, func(ctx context.Context, d *pluginsdk.ResourceData, meta interface{}) (data []*pluginsdk.ResourceData, err error) {
+			id, err := parse.FrontDoorRuleSetAssociationID(d.Id())
+			if err != nil {
+				return []*pluginsdk.ResourceData{}, err
+			}
+
+			client := meta.(*clients.Client).Cdn.FrontDoorRoutesClient
+			resp, err := client.Get(ctx, id.ResourceGroup, id.ProfileName, id.AfdEndpointName, id.AssociationName)
+			if err != nil {
+				return []*pluginsdk.ResourceData{}, fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			if resp.RouteProperties == nil {
+				return []*pluginsdk.ResourceData{}, fmt.Errorf("retrieving %s: 'RouteProperties' are 'nil'", id)
+			}
+			ruleSets := flattenRuleSetResourceArray(resp.RouteProperties.RuleSets)
+
+			d.SetId(id.ID())
+			d.Set("cdn_frontdoor_route_id", parse.NewFrontDoorRouteID(id.SubscriptionId, id.ResourceGroup, id.ProfileName, id.AfdEndpointName, id.AssociationName).ID())
+			d.Set("cdn_frontdoor_rule_set_ids", ruleSets)
+
+			return []*pluginsdk.ResourceData{d}, nil
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"cdn_frontdoor_route_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontDoorRouteID,
+			},
+
+			"cdn_frontdoor_rule_set_ids": {
+				Type:     pluginsdk.TypeSet,
+				Required: true,
+				ForceNew: true,
```

Comment:
It seems like this is updatable in the main resource, could we make it updatable here as well?